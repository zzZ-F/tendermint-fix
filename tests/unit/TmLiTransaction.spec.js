import { shallowMount } from "@vue/test-utils"
import { TmLiTransaction } from "@/index"
import transactions from "../../assets/txs.js"

describe("TmLiTransaction", () => {
  let wrapper
  let propsData = {
    devMode: true,
    transaction: transactions[0],
    address: "tb1d4u5zerywfjhxuc9nudvw"
  }

  beforeEach(() => {
    wrapper = shallowMount(TmLiTransaction, { propsData })
  })

  it("has the expected html structure", () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("should show incoming transactions", () => {
    expect(wrapper.find(".tm-li-tx").classes()).toContain("tm-li-tx-received")
  })

  it("should show outgoing transactions", () => {
    wrapper.setProps({
      transaction: transactions[1],
      address: "tb1d4u5zerywfjhxuc9nudvw"
    })
    expect(wrapper.find(".tm-li-tx").classes()).toContain("tm-li-tx-sent")
  })

  it("should show all coins of the transaction", () => {
    wrapper.setProps({
      transaction: transactions[2],
      address: "tb1d4u5zerywfjhxuc9nudvw"
    })
    expect(wrapper.vm.$el).toMatchSnapshot()

    expect(wrapper.findAll(".tx-coin").length).toBe(3)
    expect(
      wrapper
        .findAll(".tx-coin")
        .at(2)
        .html()
        .toLowerCase()
    ).toContain("mattcoins")
    expect(
      wrapper
        .findAll(".tx-coin")
        .at(2)
        .html()
    ).toContain("42")
  })
})
