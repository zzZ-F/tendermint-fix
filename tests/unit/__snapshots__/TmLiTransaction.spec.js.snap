// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiTransaction has the expected html structure 1`] = `
<div
  class="tm-li-tx tm-li-tx-received"
  devmode="true"
>
  <div
    class="tx-icon"
  >
    <i
      class="material-icons"
    >
      add_circle
    </i>
  </div>
  <div
    class="tx-container"
  >
    <div
      class="tx-element tx-coins"
    >
      <div
        class="tx-coin"
      >
        <div
          class="key"
        >
          JBCOINS
        </div>
        <div
          class="value"
        >
          1,234.00
        </div>
      </div>
    </div>
    <div>
      <!---->
      <div
        class="tx-element tx-address"
      >
        Received from tb1da6xsetjg9jxgun9wdesexv05j
      </div>
    </div>
  </div>
</div>
`;

exports[`TmLiTransaction should show all coins of the transaction 1`] = `
<div
  class="tm-li-tx tm-li-tx-received"
  devmode="true"
>
  <div
    class="tx-icon"
  >
    <i
      class="material-icons"
    >
      add_circle
    </i>
  </div>
  <div
    class="tx-container"
  >
    <div
      class="tx-element tx-coins"
    >
      <div
        class="tx-coin"
      >
        <div
          class="key"
        >
          JBCOINS
        </div>
        <div
          class="value"
        >
          1,234.00
        </div>
      </div>
      <div
        class="tx-coin"
      >
        <div
          class="key"
        >
          FABOCOINS
        </div>
        <div
          class="value"
        >
          1.00
        </div>
      </div>
      <div
        class="tx-coin"
      >
        <div
          class="key"
        >
          MATTCOINS
        </div>
        <div
          class="value"
        >
          42.00
        </div>
      </div>
    </div>
    <div>
      <!---->
      <div
        class="tx-element tx-address"
      >
        Received from tb1da6xsetjg9jxgun9wdesexv05j
      </div>
    </div>
  </div>
</div>
`;
